### DISCUSSION TITLE
How to compile the source code of O3DE from GitHub to obtain the engine? I tried but failed.

### DISCUSSION THREAD

**Discussion by Yuan797** [o3de/o3de#16917] [Help Wanted]
D:\O3DE>git clone --recurse-submodules https://github.com/o3de/o3de.git
Cloning into 'o3de'...
remote: Enumerating objects: 479678, done.
remote: Counting objects: 100% (3272/3272), done.
remote: Compressing objects: 100% (1962/1962), done.
remote: Total 479678 (delta 1875), reused 2276 (delta 1210), pack-reused 476406
Receiving objects: 100% (479678/479678), 211.64 MiB | 15.63 MiB/s, done.
Resolving deltas: 100% (367764/367764), done.
Updating files: 100% (29762/29762), done.
Filtering content: 100% (2910/2910), 2.35 GiB | 20.86 MiB/s, done.

D:\O3DE>
D:\O3DE>cd o3de && git submodule update --init --recursive

D:\O3DE\o3de>
D:\O3DE\o3de>mkdir build && cd build

D:\O3DE\o3de\build>cmake .. -DCMAKE_BUILD_TYPE=Release -DO3DE_USE_CPU=ON -DO3DE_USE_CUDA=OFF -DO3DE_USE_OPENGL=ON -DO3DE_USE_VULKAN=ON -DO3DE_USE_METAL=OFF -DO3DE_ENABLE_PYTHON=OFF -DO3DE_ENABLE_MATLAB=OFF -DO3DE_ENABLE_JNI=OFF -DCMAKE_INSTALL_PREFIX=D:/O3DE
-- Building for: Visual Studio 17 2022
-- The C compiler identification is MSVC 19.37.32824.0
-- The CXX compiler identification is MSVC 19.37.32824.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: D:/AllSoft/DevelopSoft/VS2022P/IDE/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/AllSoft/DevelopSoft/VS2022P/IDE/VC/Tools/MSVC/14.37.32822/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/zlib-1.2.11-rev5-windows
-- Using the O3DE version of the ZLIB library from E:/StudyProject/O3de_allProject/3rdParty/packages/zlib-1.2.11-rev5-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/qt-5.15.2-rev7-windows
-- Downloading package into D:/O3DE/o3de/python/runtime/python-3.10.5-rev1-windows
'D:/AllSoft/OtherSoft/CmakeSoft/bin/cmake.exe' '-E' 'tar' 'xf' 'E:/StudyProject/O3de_allProject/3rdParty/downloaded_packages/python-3.10.5-rev1-windows.tar.xz'
-- Installed And Validated package at D:/O3DE/o3de/python/runtime/python-3.10.5-rev1-windows - OK
-- Using package D:/O3DE/o3de/python/runtime/python-3.10.5-rev1-windows
-- Using Python 3.10.5 at D:/O3DE/o3de/python/runtime/python-3.10.5-rev1-windows/python/python.exe
-- Python: Getting/Checking packages listed in D:/O3DE/o3de/python/requirements.txt
-- Python: Getting/Checking packages listed in D:/O3DE/o3de/python/requirements.txt - New packages were installed
-- Python: linking D:/O3DE/o3de/Tools/LyTestTools into python site-packages...
-- Python: linking D:/O3DE/o3de/Tools/RemoteConsole/ly_remote_console into python site-packages...
-- Python: linking D:/O3DE/o3de/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools into python site-packages...
-- Python: linking D:/O3DE/o3de/scripts/o3de into python site-packages...
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/RapidJSON-1.1.0-rev1-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/Lua-5.4.4-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/RapidXML-1.13-rev1-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/zstd-1.35-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/cityhash-1.1-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/googletest-1.8.1-rev4-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/googlebenchmark-1.7.0-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/lz4-1.9.4-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/SQLite-3.37.2-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/OpenSSL-1.1.1o-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/expat-2.4.2-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/tiff-4.2.0.15-rev3-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/assimp-5.2.5-rev1-windows
-- Python: linking E:/StudyProject/O3de_allProject/3rdParty/packages/assimp-5.2.5-rev1-windows/assimp/port/PyAssimp into python site-packages...
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/xxhash-0.7.4-rev1-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/AWSNativeSDK-1.9.50-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/Crashpad-0.8.0-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/pybind11-2.10.0-rev1-multiplatform
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/d3dx12-headers-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/vulkan-validationlayers-1.3.261-rev1-windows
-- Python: linking D:/O3DE/o3de/Gems/Atom/RPI/Tools into python site-packages...
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/mcpp-2.7.2_az.2-rev1-windows
-- Downloading package into E:/StudyProject/O3de_allProject/3rdParty/packages/DirectXShaderCompilerDxc-1.7.2308-o3de-rev1-windows
'D:/AllSoft/OtherSoft/CmakeSoft/bin/cmake.exe' '-E' 'tar' 'xf' 'E:/StudyProject/O3de_allProject/3rdParty/downloaded_packages/DirectXShaderCompilerDxc-1.7.2308-o3de-rev1-windows.tar.xz'
-- Installed And Validated package at E:/StudyProject/O3de_allProject/3rdParty/packages/DirectXShaderCompilerDxc-1.7.2308-o3de-rev1-windows - OK
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/DirectXShaderCompilerDxc-1.7.2308-o3de-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/SPIRVCross-2021.04.29-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/azslc-1.8.15-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/astc-encoder-3.2-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/squish-ccr-deb557d-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/ISPCTexComp-36b80aa-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/OpenEXR-3.1.3-rev5-windows
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/openimageio-opencolorio-2.3.17-rev4-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/png-1.6.37-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/freetype-2.11.1-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/mikkelsen-1.0.0.4-windows
-- Found OpenGL: opengl32
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/pyside2-5.15.2.1-py3.10-rev6-windows
-- Python: linking E:/StudyProject/O3de_allProject/3rdParty/packages/pyside2-5.15.2.1-py3.10-rev6-windows/pyside2/lib/site-packages into python site-packages...
-- Python: Getting/Checking packages listed in D:/O3DE/o3de/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/requirements.txt
-- Python: Getting/Checking packages listed in D:/O3DE/o3de/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/requirements.txt - New packages were installed
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/AWSGameLiftServerSDK-5.0.0-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/libsamplerate-0.2.1-rev2-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/NvCloth-v1.1.6-4-gd243404-pr58-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/PhysX-4.1.2.29882248-rev5-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/poly2tri-7f0487a-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/v-hacd-2.3-1a49edf-rev1-windows
CMake Deprecation Warning at build/_deps/recastnavigation-src/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/civetweb-1.8-rev1-windows
-- Using package E:/StudyProject/O3de_allProject/3rdParty/packages/OpenMesh-8.1-rev3-windows
-- Configuring done (172.2s)
CMake Error in Code/Framework/AzToolsFramework/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/Code/Framework/AzToolsFramework/AzToolsFramework_autogen/RXS2VG5GDI_/qrc_AssetEditorResources.cpp

CMake Error in Code/Framework/AzQtComponents/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/Code/Framework/AzQtComponents/AzQtComponents_autogen/WZGSBCMU2U_/qrc_resources.cpp

CMake Error in Code/Tools/AssetProcessor/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/Code/Tools/AssetProcessor/AssetProcessor_autogen/LXJKROEMHS_/qrc_AssetProcessor.cpp

CMake Error in Gems/Atom/Tools/MaterialCanvas/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/MaterialCanvas-30301f09/Code/MaterialCanvas_autogen/L6L7XOLXOZ_/qrc_MaterialCanvas.cpp

CMake Error in Gems/Atom/Tools/AtomToolsFramework/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/AtomToolsFramework-a36b75cd/Code/AtomToolsFramework.Static_autogen/YP2RV5E7F7_/qrc_AtomToolsAssetBrowser.cpp

CMake Error in Gems/GraphCanvas/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/GraphCanvas-30ebc195/Code/GraphCanvasWidgets_autogen/3XD3IIXUY6_/qrc_GraphCanvasEditorResources.cpp

CMake Error in Gems/Atom/Tools/MaterialEditor/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/MaterialEditor-91ffad63/Code/MaterialEditor_autogen/L6L7XOLXOZ_/qrc_MaterialEditor.cpp

CMake Error in Gems/Atom/Tools/PassCanvas/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/PassCanvas-6c5a730a/Code/PassCanvas_autogen/L6L7XOLXOZ_/qrc_PassCanvas.cpp

CMake Error in Gems/Atom/Tools/ShaderManagementConsole/Code/CMakeLists.txt:
  Cannot find source file:

    D:/O3DE/o3de/build/External/ShaderManagementConsole-3cf73768/Code/ShaderManagementConsole_autogen/L6L7XOLXOZ_/qrc_ShaderManagementConsole.cpp

-- Generating done (6.9s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_BUILD_TYPE
    O3DE_ENABLE_JNI
    O3DE_ENABLE_MATLAB
    O3DE_ENABLE_PYTHON
    O3DE_USE_CPU
    O3DE_USE_CUDA
    O3DE_USE_METAL
    O3DE_USE_OPENGL
    O3DE_USE_VULKAN

CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by tarboss** [comment]
1.  git clone repo
2.  *** config:
D:\O3DE\23.05.1\cmake\runtime\bin\cmake -S D:/o3de/dev/o3de -B D:/o3de/dev/o3de/build -G "Visual Studio 16" -A x64 -DLY_3RDPARTY_PATH="D:/O3DE/3rdParty" -DLY_UNITY_BUILD=ON -DLY_PROJECTS="AutomatedTesting"
3. *** build:
D:\O3DE\23.05.1\cmake\runtime\bin\cmake --build D:/o3de/dev/o3de/build --target AutomatedTesting.GameLauncher Editor --config profile -- /m
